src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,116 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] currentTable;
+    private int currentCapacity;
+    private int threshHold;
+    private int size;
+
+    public MyHashMap() {
+        this.currentCapacity = DEFAULT_CAPACITY;
+        this.threshHold = (int) (DEFAULT_CAPACITY * DEFAULT_LOAD_FACTOR);
+        this.currentTable = (Node<K, V>[]) new Node[currentCapacity];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= threshHold) {
+            resizeTable();
+        }
+        if (Objects.isNull(currentTable[getIndexByKey(key)])) {
+            putFirstValue(key, value);
+        } else {
+            putNextValue(key, value);
+        }
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> currentNode = currentTable[getIndexByKey(key)];
+        while (hasNext(currentNode)) {
+            if (Objects.equals(key, currentNode.key)) {
+                return currentNode.value;
+            }
+            currentNode = currentNode.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void resizeTable() {
+        currentCapacity += currentCapacity;
+        threshHold = (int) (currentCapacity * DEFAULT_LOAD_FACTOR);
+        size = 0;
+        Node<K, V>[] increasedEmptyTable = (Node<K, V>[]) new Node[currentCapacity];
+        Node<K, V>[] oldTable = currentTable;
+        currentTable = increasedEmptyTable;
+        for (Node<K, V> oldNode : oldTable) {
+            if (Objects.nonNull(oldNode)) {
+                put(oldNode.key, oldNode.value);
+                Node<K, V> temp = oldNode.next;
+                while (hasNext(temp)) {
+                    put(temp.key, temp.value);
+                    temp = temp.next;
```

Comment:
I think we can simplify this method. For example: we can use while cycle with condition (oldNode != null) and if it is u will put it in your new table.
```suggestion
while(hasNext(oldNode)) {
put(oldNode.key, oldNode.value);
oldNode = oldNode.next;
```
Check my suggestion in IDE, it should work (but i am not sure :D)